fuels-contract/src/contract.rs

```diff
@@ -275,12 +276,20 @@ impl Contract {
             fuel_client: provider.client.clone(),
             datatype: PhantomData,
             output_params: output_params.to_vec(),
-            custom_inputs,
+            compute_calldata_offset,
             external_contracts: None,
             wallet: wallet.clone(),
         })
     }
 
+    // Returns true if the method call takes custom inputs or has more than one argument. This is used to determine whether we need to compute the `call_data_offset`.
+    fn should_compute_call_data_offset(args: &[Token]) -> bool {
+        match args.iter().any(|t| matches!(t, Token::Struct(_))) {
```

Comment:
Shouldn't we also check for `Enum` and in the future `Tuples`?

Reply:
Yes for enums. Tuples might or might not be a thing, we'll see once the time comes.

Reply:
yes!